### PR TITLE
AWS S3 Sink Kamelet

### PR DESCRIPTION
**Pull request by claudio4j**
#112

### CODE REVIEW

**Review by oscerd — CHANGES_REQUESTED** [review]
I think you should follow the approach we use in dropbox sink: https://github.com/apache/camel-kamelets/blob/master/dropbox-sink.kamelet.yaml#L51

And state in the description that the sink expects the header to be set.

**Comment by claudio4j** [comment]
In camel-dropbox if there is no header to designate the filename, it uses the exchange id, but in camel-aws2-s3 there is no such default for filename. So, in this case I think requiring the filename is appropriate isn't ? Or should we set the exchange id if there is no CamelAwsS3Key header ?

**Comment by oscerd** [comment]
> In camel-dropbox if there is no header to designate the filename, it uses the exchange id, but in camel-aws2-s3 there is no such default for filename. So, in this case I think requiring the filename is appropriate isn't ? Or should we set the exchange id if there is no CamelAwsS3Key header ?

I meant to say the same style as dropbox, by adding an otherwise step to set the header as exchange id. Like in the minio sink connector for example: https://github.com/apache/camel-kamelets/blob/master/minio-sink.kamelet.yaml#L60

File: aws-s3-sink.kamelet.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            steps:
+            - set-header:
+                name: CamelAwsS3Key
+                simple: "${header[ce-file]}"
+          otherwise:
+            steps:
+            - set-header:
+                name: CamelAwsS3Key
+                simple: "${exchangeId}"
+      - log: "s3 body - ${body}"
```
Comment (oscerd) [inline aws-s3-sink.kamelet.yaml:86]: I don't think this is needed, eventually a user can add a step with a log sink in the middle of a binding.

**Review by oscerd — CHANGES_REQUESTED** [review]
Just a little thing.

File: aws-s3-sink.kamelet.yaml (hunk 1 above)
Comment (claudio4j) [reply aws-s3-sink.kamelet.yaml]: Ahh, just forgot to remove it before pushing.

**Review by nicolaferraro — APPROVED** [review]

**Pull request merged**